app/gui/src/project-view/ProjectView.vue

```diff
@@ -56,7 +66,10 @@ provideSettings()
 
 const visible = ref(false)
 provideVisibility(visible)
-openedProjects.registerProject(props)
+openedProjects.registerProject({
+  ...props,
+  projectDisplayedName: readonly(toRef(props, 'projectDisplayedName')),
+})
```

Comment:
I think it would be safer to call `toRefs` on props here (changing also arg type of `registerProject` to `ToRefs<RegisterProject>`), and unref value if necessary inside.

Reply:
Yes. If you think it's a bit noisy, we may also revert it, but then I would rather make `registerProject` taking `Ref` in place of `ToValue`, so we better track what should be reactive.